bot/exts/moderation/infraction/infractions.py

```diff
@@ -664,6 +645,26 @@ async def cog_command_error(self, ctx: Context, error: Exception) -> None:
                 await ctx.send(str(error.errors[0]))
                 error.handled = True
 
+    @commands.Cog.listener()
+    async def on_member_join(self, member: Member) -> None:
+        """
+
+        Apply active timeout infractions for returning members.
+
+        This is needed for users who might have had their infraction edited in our database but not in Discord itself.
+        """
+        active_timeouts = await self.bot.api_client.get(
+            endpoint="bot/infractions",
+            params={"active": "true", "type": "timeout", "user__id": member.id}
+        )
+        if active_timeouts and not member.is_timed_out():
+            reason = f"Applying active timeout for returning member: {active_timeouts[0]['id']}"
+
+        async def action() -> None:
+            await member.edit(timed_out_until=arrow.get(active_timeouts[0]["expires_at"]).datetime, reason=reason)
```

Comment:
I get the following error when my test account that I muted rejoins the server (while the timeout is still active) because `member.is_timed_out()` is `True` and hence `reason` is not assigned a value:
```
bot-1        |   File "/bot/bot/exts/moderation/infraction/infractions.py", line 664, in action
bot-1        |     await member.edit(timed_out_until=arrow.get(active_timeouts[0]["expires_at"]).datetime, reason=reason)
bot-1        |                                                                                                    ^^^^^^
bot-1        | NameError: cannot access free variable 'reason' where it is not associated with a value in enclosing scope
```